google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java

```diff
@@ -15,39 +15,82 @@
  */
 package com.google.cloud.storage;
 
+import com.google.api.core.ApiFunction;
+import com.google.cloud.StringEnumType;
+import com.google.cloud.StringEnumValue;
+
 /**
  * Enums for the storage classes.
  * See https://cloud.google.com/storage/docs/storage-classes for details.
  */
-public enum StorageClass {
+public class StorageClass extends StringEnumValue {
+
+  private StorageClass(String constant) {
+    super(constant);
+  }
+
+  private static ApiFunction<String, StorageClass> CONSTRUCTOR =
+      new ApiFunction<String, StorageClass>() {
+        @Override
+        public StorageClass apply(String constant) {
+          return new StorageClass(constant);
+        }
+      };
+
+  private static StringEnumType<StorageClass> type = new StringEnumType(
+      StorageClass.class.getName(),
+      CONSTRUCTOR);
 
   /**
    * Regional storage class.
    */
-  REGIONAL,
+  public static StorageClass REGIONAL = type.createAndRegister("REGIONAL");
 
   /**
    * Multi-regional storage class.
    */
-  MULTI_REGIONAL,
+  public static StorageClass MULTI_REGIONAL = type.createAndRegister("MULTI_REGIONAL");
 
   /**
    * Nearline storage class.
    */
-  NEARLINE,
+  public static StorageClass NEARLINE = type.createAndRegister("NEARLINE");
 
   /**
    * Coldline storage class.
    */
-  COLDLINE,
+  public static StorageClass COLDLINE = type.createAndRegister("COLDLINE");
 
   /**
    * Standard storage class.
    */
-  STANDARD,
+  public static StorageClass STANDARD = type.createAndRegister("STANDARD");
 
   /**
    * Durable Reduced Availability (deprecated)
    */
-  DURABLE_REDUCED_AVAILABILITY,
+  public static StorageClass DURABLE_REDUCED_AVAILABILITY = type.createAndRegister(
+      "DURABLE_REDUCED_AVAILABILITY");
+
+  /**
+   * Get the StorageClass for the given String constant, and throw an exception if the constant is
+   * not recognized.
+   */
+  public static StorageClass valueOfStrict(String constant) {
```

Comment:
Can we move these methods (valueOf/valueOfStrict/values) into the parent class? It looks like they are common enough.